libraries/AP_HAL_PX4/RCInput.cpp

```diff
@@ -111,6 +119,30 @@ void PX4RCInput::clear_overrides()
     }
 }
 
+const char *PX4RCInput::input_source_name(uint8_t id) const
+{
+    switch(id) {
+    case input_rc_s::RC_INPUT_SOURCE_UNKNOWN:         return "UNKNOWN";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_PPM:      return "PX4FMU_PPM";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_PPM:       return "PX4IO_PPM";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_SPEKTRUM:  return "PX4IO_SPEKTRUM";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_SBUS:      return "PX4IO_SBUS";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_ST24:      return "PX4IO_ST24";
+    case input_rc_s::RC_INPUT_SOURCE_MAVLINK:         return "MAVLINK";
+    case input_rc_s::RC_INPUT_SOURCE_QURT:            return "QURT";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_SPEKTRUM: return "PX4FMU_SPEKTRUM";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_SBUS:     return "PX4FMU_SBUS";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_ST24:     return "PX4FMU_ST24";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_SUMD:     return "PX4FMU_SUMD";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_DSM:      return "PX4FMU_DSM";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_SUMD:      return "PX4IO_SUMD";
+    case input_rc_s::RC_INPUT_SOURCE_PX4FMU_SRXL:     return "PX4FMU_SRXL";
+    case input_rc_s::RC_INPUT_SOURCE_PX4IO_SRXL:      return "PX4IO_SRXL";
+    default:                                          return "?!";
```

Comment:
Could this say ```UNKNOWN``` or ```ERROR``` instead of ```"?!"``` My first thought if I ever saw that would be that something went wrong with the transport mechanism somewhere :)